lib/clients/jira.go

```diff
@@ -337,24 +339,32 @@ func (j realJIRAClient) UpdateComment(issue jira.Issue, id string, comment githu
 // error. If it continues to fail until a maximum time is reached, it returns
 // a nil result as well as the returned HTTP response and a timeout error.
 func (j realJIRAClient) request(f func() (interface{}, *jira.Response, error)) (interface{}, *jira.Response, error) {
+	log := j.config.GetLogger()
+
 	var ret interface{}
 	var res *jira.Response
-	var err error
+	var reqErr error
 
 	op := func() error {
-		ret, res, err = f()
-		return err
+		ret, res, reqErr = f()
+		return reqErr
```

Comment:
there is no need to create a closure here and set value on the variable in the parent scope. if backoff.RetryNotify times out after all possible retries, then it returns the last error produced by the operation. https://github.com/cenkalti/backoff/blob/master/retry.go#L42